### PR TITLE
Upgrade to RN 0.61.1

### PR DESCRIPTION
**Pull request by emin93**
# Summary

This PR contains the latest changes from the RN 0.61 template.

### CODE REVIEW

File: template/android/build.gradle [hunk 1]
```diff
@@ -6,14 +6,13 @@ buildscript {
...
         targetSdkVersion = 28
-        supportLibVersion = "28.0.0"
     }
     repositories {
         google()
         jcenter()
     }
     dependencies {
-        classpath("com.android.tools.build:gradle:3.4.1")
+        classpath("com.android.tools.build:gradle:3.4.2")
```
Comment (kaioduarte) [inline template/android/build.gradle]: Actually the version should be `3.5.0` accordingly to this commit https://github.com/facebook/react-native/commit/4763000554c71be40b82a740651706781acad2e3#diff-d1933be33ff2cf0502f4e7420581a018

Comment (emin93) [reply template/android/build.gradle]: They reverted it back according to this commit: https://github.com/facebook/react-native/commit/8750eb253858dd8b9d3480bd612b524fb38df1f1

**Comment by emin93** [comment]
Just saw, 0.61.1 got released, will include those changes as well.

File: template/android/build.gradle (hunk 1 above)
Comment (kaioduarte) [reply template/android/build.gradle]: I didn't notice that, sorry

File: template/package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
   },
   "dependencies": {
     "react": "16.8.1",
```
Comment (kaioduarte) [inline template/package.json]: React version was bumped to 16.9.0

Comment (emin93) [reply template/package.json]: That's what they wrote in the changelog, but they actually haven't done so: https://github.com/facebook/react-native/blob/v0.61.1/template/package.json

FYI: I don't add the changes hand by hand, I use the template folder from the official repo.

Comment (emin93) [reply template/package.json]: You can also check the upgrade helper and there you'll see as well that the React version hasn't been changed: https://react-native-community.github.io/upgrade-helper/?from=0.60.6&to=0.61.1

**Comment by marcorm** [comment]
Hi all, is there currently a way to create a rn 0.61.1 project with typescript template? Or should i wait for this to be released? thx a lot!

File: template/package.json [hunk 3]
```diff
@@ -3,25 +3,27 @@
...
+    "react-native": "0.61.1"
   },
   "devDependencies": {
     "@babel/core": "^7.5.0",
     "@babel/runtime": "^7.5.0",
     "@react-native-community/eslint-config": "^0.0.3",
     "@types/jest": "^24.0.18",
-    "@types/react": "^16.9.2",
-    "@types/react-native": "^0.60.14",
+    "@types/react": "^16.9.3",
```
Comment (radko93) [inline template/package.json]: Maybe we should align @types version with React version? We might end up in a state when types are incompatible with the version of React that we use

File: template/_gitignore [hunk 4]
```diff
@@ -44,6 +44,7 @@ yarn-error.log
 buck-out/
```
Comment (radko93) [inline template/_gitignore]: There's a removal of Vscode section from gitignore as well (at least when I tried to copy the template)

File: template/package.json (hunk 3 above)
Comment (emin93) [reply template/package.json]: I thought so as well, I just wasn't sure if the `@types` versions always match the library versions. But I think now I understood the version naming pattern 😄

File: template/_gitignore (hunk 4 above)
Comment (emin93) [reply template/_gitignore]: Haha, yes, the VS Code section is an addition I did to the template as most of the TS developers use VS code. I guess if the goal is to stay consistent with the default template as much as possible, I should remove it, hmm?

Comment (radko93) [reply template/_gitignore]: Ok then it's fine if you did it

**Pull request merged**